Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Confirm.hs

```diff
@@ -124,14 +132,19 @@ handler transporter req quote = do
           case routeInfo of
             Just route -> setExp (searchRequestKey $ getId ride.id) route 14400
             Nothing -> logDebug "Unable to get the key"
+          let destination = last booking.toLocation
           Esq.runTransaction $ do
             when isNewRider $ QRD.create riderDetails
             QRB.updateRiderId booking.id riderDetails.id
             QRB.updateStatus booking.id DRB.TRIP_ASSIGNED
-            QBL.updateAddress booking.fromLocation.id req.fromAddress
-            QBL.updateAddress booking.toLocation.id req.toAddress
+            QL.updateAddress booking.fromLocation.id req.fromAddress
+            QL.updateAddress destination.id req.toAddress
             whenJust req.mbRiderName $ QRB.updateRiderName booking.id
             QRide.create ride
+          mappings <- DRide.locationMappingMakerForRide ride
+          for_ mappings $ \locMap -> do
+            Esq.runTransaction $ QLocationMapping.create locMap
```

Comment:
Location mapping for ride should be created inside when we are creating ride.

Reply:
The  `locationMappingMakerForRide ` uses MonadFlow constraint hence cannot be used inside `Esq.create`.